File: python/scrabble-score/scrabble_score.py
from collections import Counter


def score(word):
    result = 0

    letter_scores = {
        "AEIOULNRST": 1,
        "DG": 2,
        "BCMP": 3,
        "FHVWY": 4,
        "K": 5,
        "JX": 8,
        "QZ": 10,
    }

    word_count = Counter(word.upper())

    for letter, total in word_count.items():
        for letter_score in letter_scores:
            if letter in letter_score:
                result += (letter_scores[letter_score] * total)

    return result

File: go/scrabble-score/scrabble_score.go
package scrabble

import (
	"strings"
	"unicode"
)

var wordList = map[string]int{
	"AEIOULNRST": 1,
	"DG":         2,
	"BCMP":       3,
	"FHVWY":      4,
	"K":          5,
	"JX":         8,
	"QZ":         10,
}

func Score(word string) int {
	var (
		cache      = map[rune]int{}
		result int = 0
	)

	for _, letter := range word {

		letter = unicode.ToUpper(letter)

		value, ok := cache[letter]
		if ok {
			result += value
			continue
		}

		for k, v := range wordList {
			if strings.ContainsRune(k, letter) {
				cache[letter] = v
				result += v
				break

			}
		}

	}

	return result
}
